components/admin_console/permission_schemes_settings/guest_permissions_tree/guest_permissions_tree.jsx

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {FormattedMessage} from 'react-intl';
+
+import PermissionGroup from '../permission_group.jsx';
+
+import EditPostTimeLimitButton from '../edit_post_time_limit_button';
+import EditPostTimeLimitModal from '../edit_post_time_limit_modal';
+
+export default class GuestPermissionsTree extends React.Component {
+    static propTypes = {
+        scope: PropTypes.string.isRequired,
+        role: PropTypes.object.isRequired,
+        onToggle: PropTypes.func.isRequired,
+        parentRole: PropTypes.object,
+        selected: PropTypes.string,
+        selectRow: PropTypes.func.isRequired,
+        readOnly: PropTypes.bool,
+    };
+
+    static defaultProps = {
+        role: {
+            permissions: [],
+        },
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            editTimeLimitModalIsVisible: false,
+        };
+
+        this.ADDITIONAL_VALUES = {
+            edit_post: {
+                editTimeLimitButton: <EditPostTimeLimitButton onClick={this.openPostTimeLimitModal}/>,
+            },
+        };
+
+        this.permissions = [
+            'create_private_channel',
```

Comment:
This can be references to the `Permissions` set of constants

Reply:
Done!